File: src/medium/t6/solve.go
package main

import "math"

// 双指针
func minSubArrayLen(target int, nums []int) int {
	// 头指针 尾指针
	var start, end = 0, 0
	// 记录头指针和尾指针区域的和
	var sum = 0
	// 记录最小长度
	var size = math.MaxInt
	// 尾指针走到最后为止
	for end < len(nums) {
		sum += nums[end]
		// 如果满足区域和>=target
		for sum >= target {
			// 计算区域距离
			size = min(size, end-start+1)
			// 头指针前进一位，区域和减去头指针位置
			sum -= nums[start]
			start++
		}
		end++
	}
	if size == math.MaxInt {
		size = 0
	}
	return size
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: src/medium/t6/solve.rs
use std::{cmp::min, i32::MAX};

pub fn min_sub_array_len(target: i32, nums: Vec<i32>) -> i32 {
    let mut start = 0_usize;
    let mut size = MAX;
    let mut sum = 0;
    for end in 0..nums.len() {
        sum += nums[end];
        while sum >= target {
            size = min(size, (end - start + 1) as i32);
            sum -= nums[start];
            start += 1;
        }
    }
    if size == MAX {
        size = 0
    }
    size
}
